R/get_tiles.R

```diff
@@ -0,0 +1,248 @@
+#' @title Retrieve Macrostrat vectorized tiles
+#'
+#' @description A function to retrieve map tile data from the
+#'   [Macrostrat tile server](https://tiles.macrostrat.org/). This includes
+#'   geologic line features such as faults, anticlines, and moraines, as well as
+#'   geologic map polygons (similar to [get_map_outcrop()] but for an entire
+#'   geographic tile). The function retrieves the data in vectorized (\code{sf})
+#'   tile format. Note that the geographic features come from the "carto"
+#'   Macrostrat scale, which is primarily for visualization purposes, and makes
+#'   many assumptions about the relative priority of each source map. The layers
+#'   and scales are seamlessly blended based on the chosen \code{zoom} level, so
+#'   no scale-dependent decisions can or should be made.
+#'
+#' @details The tile indices (\code{x} and \code{y}) are zero-indexed, meaning
+#'   that the first tile in each dimension is \code{0}. The zoom level is also
+#'   zero-indexed, meaning that the first (most zoomed out) zoom level is
+#'   \code{0}. For a given zoom level, the possible tile indices are \code{0} to
+#'   \code{2 ^ zoom - 1}. Note that retrieving multiple (or all tiles) at higher
+#'   zoom levels returns large (memory-wise) data objects.
+#'
+#'   Also note that two (or more) tiles may contain parts of the same geologic
+#'   outcrop polygon. Since tiles also overlap slightly, these separate outcrop
+#'   components may also overlap, which may lead to confusing visualizations or
+#'   invalid downstream analyses. To address this, users are encouraged to union
+#'   polygons that have the same \code{map_id} values using the [sf::st_union()]
+#'   function (either with a combination of [dplyr::group_by()] and
+#'   [dplyr::summarise()] or with [palaeoverse::group_apply()], the former of
+#'   which is significantly faster for a large number of tiles). We have not
+#'   implemented this here to reduce the number of package dependencies and to
+#'   reduce the computational time of this function.
+#'
+#' @param zoom \code{integer}. The zoom level of the tile(s) to retrieve. The
+#'   minimum zoom level is \code{0}.
+#' @param x \code{integer}. The x index/indices of the tile(s) to retrieve. If
+#'   \code{NULL} (the default), tiles of all x indices will be retrieved and
+#'   combined.
+#' @param y \code{integer}. The y index/indices of the tile(s) to retrieve. If
+#'   \code{NULL} (the default), tiles of all y indices will be retrieved and
+#'   combined.
+#' @param combined \code{logical}. Whether all tiles should be combined. If
```

Comment:
After playing around with the function more, I'm not sure there is much value in having control over this argument? I feel that this should just be the behaviour. I think if we wanted to entertain the idea of including any additional arguments, we could consider including two logical ones (`units` and `lines`) controlling which data are returned, defaulting to TRUE for both(?). 

We could also consider whether outputting the data as a list is the best option? It could also be one sf data.frame with a column specifying whether they are units or lines. This way users can directly call `plot(output)`.... just a thought for contemplation!

Reply:
I kept this argument in because combining everything takes a long time and I thought it might be nice for users to be able to skip that. However, I'm not sure there are many use cases for this (maybe if the users wants to filter the polygons in some way before combining them).

As far as the output format, I'm not sure I support combining the two types of data into a single sf data.frame, but I'm totally in favor of allowing users to customize which of the types types of data they get back.